Tests/WpfTest/MainWindow.xaml.cs

```diff
@@ -16,36 +19,92 @@
 using System.Windows.Media.Imaging;
 using System.Windows.Navigation;
 using System.Windows.Shapes;
+using WpfTest.Data;
 using static WpfTest.Services.EmailSendServiceClass;
 
 namespace WpfTest
 {
     /// <summary>
     /// Interaction logic for MainWindow.xaml
     /// </summary>
-    public partial class MainWindow : Window
+    public partial class MainWindow
     {
-        public MainWindow()
+        public MainWindow() => InitializeComponent();
+
+        List<string> result;
+
+        private async void LoadButton_Click(object sender, RoutedEventArgs e)
         {
-            InitializeComponent();
+            var openDialog = new OpenFileDialog
+            {
+                Title = "Открыть CSV-файл",
+                Filter = "Csv файлы (*.csv)|*.csv|Все файлы (*.*)|*.*",
+                InitialDirectory = Environment.CurrentDirectory
+            };
+
+            if (openDialog.ShowDialog() == false) return;
+            
+            string fileName = openDialog.FileName;
+            if (File.Exists(fileName) == false) return;
+
+            result = await Task.Run(() => LoadFile(fileName)).ConfigureAwait(true);
+
+            StatusTextBlock.Text = "Done!";
+            LoadButton.IsEnabled = false;
+            SaveButton.IsEnabled = true;
         }
 
-        private void Button_Click(object sender, RoutedEventArgs e)
+        private async void SaveButton_Click(object sender, RoutedEventArgs e)
         {
+            var saveDialog = new SaveFileDialog
+            {
+                Title = "Сохранить как TXT-файл",
+                Filter = "Txt файлы (*.txt)|*.txt|Все файлы (*.*)|*.*",
+                InitialDirectory = Environment.CurrentDirectory,
+                OverwritePrompt = true,
+            };
+
+            if (saveDialog.ShowDialog() == false) return;
```

Comment:
```suggestion
            if (saveDialog.ShowDialog() != true) return;
```
Если пользователь нажмёт крестик в правом верхнем углу, то вернётся null.